### PR TITLE
fix(Init): raise InitFailedError on keyboard interrupt on pre-commit …

### PR DESCRIPTION
**Pull request by bearomorphism**
…hook question, remove unreachable code path

## Checklist

- [x] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [ ] Add test cases to all the changes you introduce
- [ ] Run `poetry all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [ ] Verify the feature/bug fix works as expected in real-world scenarios
  - [ ] Test edge cases and error conditions
  - [ ] Ensure backward compatibility is maintained
  - [ ] Document any manual testing steps performed
- [ ] Update the documentation for the changes

### Documentation Changes

- [x] Run `poetry doc` locally to ensure the documentation pages renders correctly
- [x] Check and fix any broken links (internal or external) in the documentation

> When running `poetry doc`, any broken internal documentation links will be reported in the console output like this:
>
> ```text
> INFO    -  Doc file 'config.md' contains a link 'commands/bump.md#-post_bump_hooks', but the doc 'commands/bump.md' does not contain an anchor '#-post_bump_hooks'.
> ```

### CODE REVIEW

File: commitizen/commands/init.py [hunk 1]
```diff
@@ -369,17 +374,3 @@ def _get_config_data(self) -> dict[str, Any]:
...
-        config_data = self._get_config_data()
-        with smart_open(
-            self._PRE_COMMIT_CONFIG_PATH, "w", encoding=self.encoding
-        ) as config_file:
-            yaml.safe_dump(config_data, stream=config_file)
-
-        if not self.project_info.is_pre_commit_installed:
-            raise InitFailedError("pre-commit is not installed in current environment.")
-        if hook_types is None:
-            hook_types = ["commit-msg", "pre-push"]
```
Comment (bearomorphism) [inline commitizen/commands/init.py:383]: This branch is redundant. `hook_types` is never `None` as we already checked it at callsite.

File: commitizen/commands/init.py [hunk 2]
```diff
@@ -161,20 +180,6 @@ def __call__(self) -> None:
...
-            choices=[
-                questionary.Choice("commit-msg", checked=False),
-                questionary.Choice("pre-push", checked=False),
-            ],
-        ).unsafe_ask()
-        if hook_types:
-            try:
-                self._install_pre_commit_hook(hook_types)
-            except InitFailedError as e:
-                raise InitFailedError(f"Failed to install pre-commit hook.\n{e}")
```
Comment (bearomorphism) [inline commitizen/commands/init.py:176]: This try catch is also unnecessary

**Review by woile — APPROVED** [review]

**Comment by bearomorphism** [comment]
Just adding a note that I have confirmed the issue with @Lee-W offline

File: commitizen/commands/init.py [hunk 3]
```diff
@@ -321,26 +338,6 @@ def _ask_update_changelog_on_bump(self) -> bool:
...
-                f"Error running {cmd_str}."
-                "Outputs are attached below:\n"
-                f"stdout: {c.out}\n"
-                f"stderr: {c.err}"
-            )
-            raise InitFailedError(err_msg)
-
-    def _gen_pre_commit_cmd(self, hook_types: list[str]) -> str:
-        """Generate pre-commit command according to given hook types"""
-        if not hook_types:
```
Comment (bearomorphism) [inline commitizen/commands/init.py:338]: This `not hook_types` is again redundant.

File: commitizen/commands/init.py [hunk 4]
```diff
@@ -150,9 +150,40 @@ def __call__(self) -> None:
...
+            if not self.project_info.is_pre_commit_installed:
+                raise InitFailedError("Failed to install pre-commit hook.")
+
+            cmd_str = "pre-commit install " + " ".join(
+                f"--hook-type {ty}" for ty in hook_types
+            )
+            c = cmd.run(cmd_str)
+            if c.return_code != 0:
+                raise InitFailedError(
+                    "Failed to install pre-commit hook.\n"
```
Comment (bearomorphism) [inline commitizen/commands/init.py:182]: Just added this line to keep the original behavior.

File: commitizen/commands/init.py [hunk 5]
```diff
@@ -150,9 +150,43 @@ def __call__(self) -> None:
...
+            config_data = self._get_config_data()
+            with smart_open(
+                self._PRE_COMMIT_CONFIG_PATH, "w", encoding=self.encoding
+            ) as config_file:
+                yaml.safe_dump(config_data, stream=config_file)
+
+            if not self.project_info.is_pre_commit_installed:
+                raise InitFailedError(
+                    "Failed to install pre-commit hook.\n"
+                    "pre-commit is not installed in current environment."
```
Comment (bearomorphism) [inline commitizen/commands/init.py:173]: This line is also added to keep the original behavior.

File: commitizen/commands/init.py (hunk 2 above)
Comment (bearomorphism) [reply commitizen/commands/init.py:176]: I mean it is not easy to trace the error. This try-catch inserts the error message "Failed to install pre-commit hook.\n" at the beginning of the inner error message.

**Comment by bearomorphism** [comment]
Added some more changes related to `hook_types` to reduce complexity. Thanks.
